### PR TITLE
Component color variables

### PR DESCRIPTION
**Pull request by b-kelly**
Allows us to set custom colors for certain components on an opt-in basis. Useful for themes where you want your (for example) buttons to deviate from using secondary colors to something else.

Additionally, this allows us to completely remove the need for our hacky `_stacks-configuration-*` files and `#stacks-internals #load-config()` calls.

### CODE REVIEW

File: lib/css/base/_stacks-configuration-dynamic.less [hunk 1]
```diff
@@ -21,38 +21,9 @@
...
-    @button-filled-selected-color:                   var(--theme-secondary-900);
-    @button-filled-selected-background-color:        var(--theme-secondary-300);
-    @button-filled-selected-border-color:            var(--theme-secondary-500);
-
-    //  Button Outline
-    @button-outlined-border-color:                   var(--theme-secondary-300);
-    @button-outlined-selected-border-color:          var(--theme-secondary-400);
+    @body-site-width:                                1060px; // TODO not overridden in prod
 
+    // TODO not overridden in prod
```
Comment (aaronshekey) [inline lib/css/base/_stacks-configuration-dynamic.less]: Glad all these TODOs are identified. I wonder if these should actually be `// Deprecated` comments? Or, honestly, I'd be happy to just yeet these straight away.

File: lib/css/components/_stacks-buttons.less [hunk 2]
```diff
@@ -16,13 +16,11 @@
 //  $   BASE STYLE
 //  ----------------------------------------------------------------------------
 & {
-    #stacks-internals #load-config();
```
Comment (aaronshekey) [inline lib/css/components/_stacks-buttons.less:19]: Absolutely thrilled to see these going.

File: lib/css/base/_stacks-configuration-dynamic.less [hunk 3]
```diff
@@ -96,16 +72,7 @@
...
-    @button-primary-hover-color:                     @white;
-    @button-primary-hover-background-color:          var(--theme-secondary-500);
-    @button-primary-active-background-color:         var(--theme-secondary-700);
-    @button-primary-selected-color:                  var(--white);
-    @button-primary-selected-background-color:       var(--theme-secondary-700);
-    @button-primary-number-color:                    var(--theme-secondary-900);
-
+    // TODO not overridden in prod
     //  Tags
     @tags-border:                                    transparent;
```
Comment (aaronshekey) [inline lib/css/base/_stacks-configuration-dynamic.less:77]: When exploring better theming on tags, I ended up deleting these. #717. I could see this entire file going away.

**Review by aaronshekey — APPROVED** [review]
I think this approach looks great. I ran this through Backstop and it didn't catch anything unexpected. I made a note or two about things we could do _now_ instead of later.

File: lib/css/base/_stacks-configuration-dynamic.less (hunk 1 above)
Comment (b-kelly) [reply lib/css/base/_stacks-configuration-dynamic.less]: Changed to `// DEPRECATED` comments. I'm going to come back in a follow up PR to yeet these entirely once I make triple sure these are not used anywhere (I'm 90% sure they aren't though).

File: lib/css/base/_stacks-configuration-dynamic.less (hunk 3 above)
Comment (b-kelly) [reply lib/css/base/_stacks-configuration-dynamic.less:77]: Nice! I'm going to leave these in at the moment so you don't have to solve a merge conflict in your PR. Happy to see these are going away early.

**Review by aaronshekey — APPROVED** [review]

**Pull request merged**